packages/eslint-plugin/src/rules/no-unsafe-call.ts

```diff
@@ -74,6 +75,52 @@ export default createRule<[], MessageIds>({
             type: isErrorType ? '`error` type' : '`any`',
           },
         });
+        return;
+      }
+
+      if (isBuiltinSymbolLike(services.program, type, 'Function')) {
+        // this also matches subtypes of `Function`, like `interface Foo extends Function {}`.
+        //
+        // For weird TS reasons that I don't understand, these are
+        //
+        // safe to construct if:
+        // - they have at least one call signature _that is not void-returning_,
+        // - OR they have at least one construct signature.
+        //
+        // safe to call (including as template) if:
+        // - they have at least one call signature
+        // - OR they have at least one construct signature.
+
+        const constructSignatures = type.getConstructSignatures();
+        if (constructSignatures.length > 0) {
+          return;
```

Comment:
[Testing] Missing coverage here. I got it with:

```ts
      code: `
interface StillUnsafe extends Function {
  property: string;
  new(): StillUnsafe;
}
declare const unsafe: StillUnsafe;
unsafe();
      `,
```

...but haven't checked if there's a simpler case.